examples/probes/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+    "flag"
+    "log"
+
+    "github.com/DNS-OARC/ripeatlas"
+    "github.com/DNS-OARC/ripeatlas/request"
+)
+
+var all bool
+var page int
+var file bool
+
+func init() {
+    flag.BoolVar(&all, "all", false, "list all probes")
+    flag.IntVar(&page, "page", 1, "pagination page to load (default first, only used without -file and arguments)")
+    flag.BoolVar(&file, "file", false, "arguments given are files to read (default probe ids to query for over HTTP)")
+}
+
+func print(p *request.Probe) {
+    log.Printf("%d %s %s/%s %s", p.Id(), p.Type(), p.AddressV4(), p.AddressV6(), p.Description())
+}
+
+func main() {
+    flag.Parse()
+
+    var a ripeatlas.Atlaser
+    if file {
+        a = ripeatlas.NewFile()
+    } else {
+        a = ripeatlas.NewHttp()
+    }
+
+    if !file && all {
+        page = 1
+        for {
+            probes, err := a.Probes(ripeatlas.Params{
+                "page": int64(page),
+            })
+            if err != nil {
+                log.Fatalf(err.Error())
+            }
+
+            got := 0
+            for p := range probes {
+                got++
+                if p.ParseError != nil {
+                    log.Println(p.ParseError.Error())
+                    break
+                }
+                print(p)
+            }
+            if got == 0 {
+                break
+            }
+            page++
+        }
```

Comment:
My thoughts were to handle pagination inside the library. I would expect library to hide pagination details from users and give them directly what they care for more (data points). But if you think otherwise I am cool :)

Reply:
There are a few reasons why I did this:
- Giving all limits options and may load API more then needed, now the user can choose
- The pagination use by Atlas is not portable, would have been a lot better to use something like Facebook has ([Cursor-based Pagination](https://developers.facebook.com/docs/graph-api/using-graph-api/#paging))
- It is now really easy to extend the library to take options like `all`, `count` etc